lib/EnzymeCore/src/EnzymeCore.jl

```diff
@@ -79,6 +79,12 @@ end
 
 Like [`Duplicated`](@ref), except also specifies that Enzyme may avoid computing
 the original result and only compute the derivative values.
+
+This should only be used if `x` is a write-only variable. Otherwise, if the differentiated
+function stores values in `x` and reads them back in subsequent computations, using
+`DuplicatedNoNeed` may result in incorrect derivatives. In particular, `DuplicatedNoNeed`
+should not be used for preallocated workspace, even if the user might not care about its
+final value.
```

Comment:
to the last sentence add "as marking a variable as NoNeed means that reads from the variable are now undefined."